update-version-h.sh

```diff
@@ -90,7 +93,7 @@ cat > libtransmission/version.h.new << EOF
 #define SHORT_VERSION_STRING      "${user_agent_prefix}"
 #define LONG_VERSION_STRING       "${user_agent_prefix} (${vcs_revision})"
 #define VERSION_STRING_INFOPLIST  ${user_agent_prefix}
-#define BUILD_STRING_INFOPLIST    14714.${major_version}.${minor_version}.${patch_version}
+#define BUILD_STRING_INFOPLIST    $((major_version + 14714)).${minor_version}.$((patch_version * 100 + release_number * 99 + beta_number))
```

Comment:
This line doesn't seem to be used anymore -- can it be removed?

Reply:
This line is the critical one when building with Xcode: it defines BUILD_STRING_INFOPLIST, which is used for CFBundleVersion in `macosx/info.plist`